packages/nuxi/src/commands/dev.ts

```diff
@@ -44,20 +44,22 @@ export default defineNuxtCommand({
 
     await setupDotenv({ cwd: rootDir, fileName: args.dotenv })
 
+    const { loadNuxt, loadNuxtConfig, buildNuxt } = await loadKit(rootDir)
+
+    const config = await loadNuxtConfig({})
+
     const listener = await listen(serverHandler, {
       showURL: false,
       clipboard: args.clipboard,
       open: args.open || args.o,
-      port: args.port || args.p || process.env.NUXT_PORT,
-      hostname: args.host || args.h || process.env.NUXT_HOST,
-      https: args.https && {
+      port: args.port || args.p || process.env.NUXT_PORT || config.devServer.port,
+      hostname: args.host || args.h || process.env.NUXT_HOST || config.devServer.host,
+      https: (args.https && {
         cert: args['ssl-cert'],
         key: args['ssl-key']
```

Comment:
```suggestion
        cert: args['ssl-cert'] || config.devServer.cert,
        key: args['ssl-key'] || config.devServer.key
```

Reply:
It seems `config.devServer.https.cert` and `config.devServer.https.key` are not directly accessible without casting.

Is the following acceptable?
```TypeScript
    const listener = await listen(serverHandler, {
      showURL: false,
      clipboard: args.clipboard,
      open: args.open || args.o,
      port: args.port || args.p || process.env.NUXT_PORT || config.devServer.port,
      hostname: args.host || args.h || process.env.NUXT_HOST || config.devServer.host,
      https: args.https && ({
        cert: args['ssl-cert'],
        key: args['ssl-key']
      } || config.devServer.https)
    })
```

That should fix the issue that HTTPS is always enabled if the configuration exists in the Nuxt config.
With that said, should we add an option to enable HTTPS from the Nuxt config as well?
I assumed with `devServer.https` being `false | { key: string, cert: string` it would default to true if a HTTPS configuration exists.

Reply:
We can disable https if `config.devServer.https === false` in check

Reply:
I think if user passes `--https` that should override nuxt.config. So logic for https piece should likely be something like:

```js
      https: (args.https || config.devServer.https)
        ? {
            cert: args['ssl-cert'] || (config.devServer.https && config.devServer.https.cert) || undefined,
            key: args['ssl-key'] || (config.devServer.https && config.devServer.https.key) || undefined
          }
        : false
    })
```

Reply:
Currently it does work like that as I made sure to test all terminal arguments override the nuxt config. I put the `--https`, `--ssl-cert` and `--ssl-key` first so that terminal arguments had priority, as I did with the host and port variables. It uses `config.devServer.https` if terminal arguments are not specified. This would just be a more verbose way of implementing it, which in my opinion isn't required to meet the same functionality.

Reply:
As currently written, it wouldn't work if the user passes `--https` but has the key set in the config.

We might make it simpler with `defu` which can merge objects for us.

Reply:
I have just tested this again and when specifying just `--https`, listhen uses its own certificate rather than the options in the nuxt config. If the user doesn't specify any terminal arguments it will use the nuxt config options. 

If the developer would like to use the options in the nuxt config, I believe they should just remove the `--https` arguments as specifying HTTPS options in nuxt config enables HTTPS anyway.

Reply:
I understand that that makes sense. But I think for a minimal extra amount of code here we handle some edge cases that might otherwise confuse people. Otherwise we'll need to document that if you pass `--https` it will override your nuxt config, and if you only pass a key or cert it will be ignored, even if you have ssl entabled in your nuxt config.

Reply:
No worries, I'll fix this up now.

Reply:
thanks - I completely understand ❤️